test/utils/custom_tsv_parser.py

```diff
@@ -0,0 +1,50 @@
+import csv
+
+
+class ParseTSVFile:
+    def __init__(self, file_path):
+        self.file_path = file_path
+
+    def read_file(self):
+        with open(self.file_path, "r", newline="", encoding="utf-8") as file:
+            reader = csv.DictReader(file, delimiter="\t")
+            return reader
+
+    def get_columns(self):
+        with open(self.file_path, "r", newline="", encoding="utf-8") as file:
```

Comment:
It would be nice not to open and close files everywhere in the code; this means that the testing will be pretty cumbersome. Not to mention the repetitions. This also leads to multiple opportunities for executing files all over the code more open doors for executing malicious file. I would like to know the rationale behind it. IO should be done only once as a data input, not wherever we can. Also, since the data size is small, we can load it in memory and pass it along so we don't have to worry about th reader being exhausted. 

I think this TSV parser could be reformatted if we really need one; otherwise, just a couple of functions would have done the job.

```python

class TsvFileParser:
    def __init__(self, file_path):
        self.file_path = file_path
        self.parsed_data :DictReader = None
    
    def parse(self, ..) -> None:
        try:
         with open(self.file_path, "r", newline="", encoding="utf-8") as file:
            reader = csv.DictReader(file, delimiter="\t")
        except YourException as e:
            # log, exit depending on the need, and inform the user or raise with custom exception handled elsewhere. But logging, exiting, and telling the user should be enough here.

        data = []
        for row in reader:
            data.append(row)
        self.parsed_data = data
        return self






tsv_parser = TsvFileParser(lovely_file_name)
parsed_data = tsv_parser.parse().data
or
tsv_parser.parse()
parsed_data = tsv_parser.parsed_data